### ISSUE TITLE
Python documentation error

### ISSUE THREAD

**Issue by TELSER1** [databricks/spark-redshift#56]
I believe you have accidentally copied the Scala syntax into the Python code examples,e.g.:

```python
df.write \
  .format("com.databricks.spark.redshift")
  .option("url", "jdbc:redshift://redshifthost:5439/database?user=username&password=pass") \
  .option("dbtable" -> "my_table_copy") \
  .option("tempdir" -> "s3://path/for/temp/data") \
  .mode("error")
  .save()
```
Presumably, all of the -> should be replaced with =, and the quotes on the assignment removed, i.e.:
```python
df.write \
  .format("com.databricks.spark.redshift")
  .option("url", "jdbc:redshift://redshifthost:5439/database?user=username&password=pass") \
  .option(dbtable="my_table_copy") \
  .option(tempdir="s3://path/for/temp/data") \
  .mode("error")
  .save()
```

However, this doesn't quite work either; it returns the error "option() got an unexpected keyword argument 'dbtable'" instead.

**Comment by JoshRosen** [comment]
Here's the correct syntax:

```python
df.write \
  .format("com.databricks.spark.redshift")
  .option("url", "jdbc:redshift://redshifthost:5439/database?user=username&password=pass") \
  .option("dbtable", "my_table_copy") \
  .option("tempdir", "s3://path/for/temp/data") \
  .mode("error")
  .save()
```

There's a fix for this as part of another PR that hasn't been merged yet. I'll just push these fixes as a separate commit.

**Comment by TELSER1** [comment]
Thanks; this resolves the error, although I promptly hit "Failed to load class for data source: com.databricks.spark.redshift" right afterwards.  This may be because I'm missing a dependency.

**Comment by JoshRosen** [comment]
How did you add `spark-redshift` to your Spark environment?

Try using the `--packages` spark-submit option, e.g.

```
$SPARK_HOME/bin/pyspark --packages databricks:spark-redshift:0.4.0-hadoop2
```

**Comment by JoshRosen** [comment]
Oh, I just realized that you're probably building your own version off of master since you're trying to use the save feature.

In this case, try publishing locally using SBT, then update the version to `0.4.1-SNAPSHOT` in your `--packages` command.

**Comment by TELSER1** [comment]
No, I was using the command you suggested.

**Comment by JoshRosen** [comment]
To clarify: you need to add this library as a dependency to your job. There are instructions at http://spark-packages.org/package/databricks/spark-redshift; this is what I was asking about.

**Comment by TELSER1** [comment]
More specifically, I am currently initializing the shell as follows
```unix
pyspark --packages databricks:spark-redshift:0.4.0-hadoop2,com.databricks:spark-csv_2.10:1.2.0
```
Spark-csv works fine; both seem to download and install.

**Comment by JoshRosen** [comment]
Ah, I see the problem:

spark-redshift 0.4.0 does not include data source support; that feature is only present in the master branch of this repository and hasn't been included in a published release yet.  To see the documentation for the version that you're using, see https://github.com/databricks/spark-redshift/tree/v0.4.0.

We are tentatively planning to publish v0.5.0 this Friday; that version will include the data sources feature. In the meantime, you'll have to build the library yourself if you want to test out its data sources support.

**Comment by TELSER1** [comment]
Alright, thanks for the update.  I can wait on the release.

**Comment by TELSER1** [comment]
Hello,

I got around to trying this out again, and running

 pyspark --packages databricks:spark-redshift:0.5.0-hadoop2,com.databricks:spark-csv_2.10:1.2.0

now throws

		::::::::::::::::::::::::::::::::::::::::::::::

		::          UNRESOLVED DEPENDENCIES         ::

		::::::::::::::::::::::::::::::::::::::::::::::

		:: databricks#spark-redshift;0.5.0-hadoop2: not found

		::::::::::::::::::::::::::::::::::::::::::::::
Is it not fully released yet, or am I misspecifying something?

**Comment by JoshRosen** [comment]
It is fully-released. Remove the `-hadoop2` part from your `--packages` command; we now publish a single artifact which is compatible with both Hadoop 1.x and 2.x.

**Comment by TELSER1** [comment]
pyspark --packages databricks:spark-redshift:0.5.0,com.databricks:spark-csv_2.10:1.2.0 yields the same message:
		::::::::::::::::::::::::::::::::::::::::::::::

		::          UNRESOLVED DEPENDENCIES         ::

		::::::::::::::::::::::::::::::::::::::::::::::

		:: databricks#spark-redshift;0.5.0: not found

		::::::::::::::::::::::::::::::::::::::::::::::

**Comment by brkyvz** [comment]
The package hasn't been published to Spark Packages yet, therefore the shorthand coordinate doesn't work. Could you please try
`com.databricks:spark-redshift_2.10:0.5.0`

**Comment by brkyvz** [comment]
@JoshRosen Maybe you could also publish the release to Spark Packages.

**Comment by JoshRosen** [comment]
@brkyvz, yep, I just realized that we haven't done that yet. Going to work on it shortly.

**Comment by TELSER1** [comment]
com.databricks:spark-redshift_2.10:0.5.0 didn't throw an error.

**Comment by JoshRosen** [comment]
We've just published 0.5.0 to spark-packages, in case you want to give it another try.

**Comment by TELSER1** [comment]
That also currently works, although I hit 

<code>java.lang.ClassNotFoundException: com.amazon.redshift.jdbc4.Driver
</code>

immediately; I presume this is because I haven't figured out how to properly provide a JDBC driver, as per the installation instructions.

**Comment by JoshRosen** [comment]
@TELSER1, try downloading Amazon's JDBC4 driver from http://docs.aws.amazon.com/redshift/latest/mgmt/configure-jdbc-connection.html, then add it to your Spark cluster via --jars.

**Comment by TELSER1** [comment]
Thanks for all the help; I had been trying to use JDBC4.1.

**Comment by JoshRosen** [comment]
@TELSER1, glad that we were able to figure everything out. I think that you technically can use the JDBC 4.1 driver if you've configured the `jdbcdriver` option to be `com.amazon.redshift.jdbc41.Driver`. I think that this confusion is common enough that it would be a good idea for `spark-redshift` to automatically use either the 4.0 or 4.1 driver depending on which is available. I've filed #83 so that we remember to follow up on this.
